loading.bs

```diff
@@ -331,6 +376,322 @@ Note: Applying the signed exchange's response here has the effect of letting a
+    1. Let |corsAttributeState| be the current state of the |el|'s
+        <{link/crossorigin}> content attribute.
+    1. Let |request| be the result of [=creating a potential-CORS request=]
+        given |url|, the empty string, and |corsAttributeState|.
+    1. Run the [=linked resource fetch setup steps=], given |el| and |request|.
+        If the result is false, then continue.
+    1. If |request|'s [=request/URL=] is same as the same as the [=alternate
+        signed exchange link=]'s [=Link Context=], return true.
+1. Return false.
+
+</ol>
+
+<div algorithm="process the alternate signed exchange linked resource">
+
+The [=process the linked resource=] steps for an [=alternate signed exchange
+link=] , given boolean |success|, and [=response=] |response|:
+
+1. If |success| is false, return.
+1. If the |response| [=response/came from a signed exchange=], create an
+    [=exchange=] with the URL of the <{link/rel/allowed-alt-sxg}> link's [=Link
+    Target=] URL and the result of [=response/cloning=] the inner response of
+    the [=alternate signed exchange=], and store the exchange to the
+    [=Document/prefetched subresource signed exchanges=] of the document.
+
+</div>
+
+<h3 algorithm id="mp-document">Monkeypatch Document</h3>
+
+At the end of
+[the Document object](https://html.spec.whatwg.org/multipage/dom.html#the-document-object)
+section, add the following lines:
+
+<blockquote dfn-for="Document">
+
+The {{Document}} has a set of <dfn dfn>prefetched signed exchanges for
+navigation</dfn>, which is a set of [=exchanges=], initially empty.  <span
+class="note">This might merge with the set of prefetched resources when
+<{link/rel/prefetch}> is specified more completely.</span>
+
+The {{Document}} has a set of <dfn dfn>prefetched
+subresource signed exchanges</dfn>, which is a set of [=exchanges=], initially
+empty.
+
+The {{Document}} has a set of <dfn dfn>transferred
+subresource signed exchanges</dfn>, which is a set of [=exchanges=], initially
+empty.
+
+The {{Document}} has a set of <dfn dfn>allowed
+alternate subresource signed exchange links</dfn>, which is a set of
+<{link/rel/allowed-alt-sxg}> links, initially empty.
+
+</blockquote>
+
+<h3 algorithm id="mp-navigation-params">Monkeypatch navigation params struct</h3>
+
+In the [=navigation params=], add the following items:
+
+<dl dfn-for="navigation params">
+
+: <dfn>prefetched subresource signed exchanges</dfn>
+:: A set of [=exchanges=], initially empty.
+: <dfn>allowed alternate subresource signed exchange links</dfn>
+:: A set of <{link/rel/allowed-alt-sxg}> links, initially empty.
+
+</dl>
+
+<h3 algorithm id="mp-fetching-the-resource-hint-link">Monkeypatch Fetching the resource hint link</h3>
+
+At the end of the [Fetching the resource hint
+link](https://w3c.github.io/resource-hints/#fetching-the-resource-hint-link)
+section in [[resource-hints]], add the following lines:
+
+The user agent SHOULD run the following steps if the response of a
```

Comment:
I think this should mokeypatch Fetch or HTML (if this is only meant for prenavigates) directly.
So when either processes a response, they need to call an algorithm defined here that would:
* Check if the response is a prefetch one (e.g. by checking its request's initiator) and otherwise bail
* Check if the response came from an SXG. Otherwise bail
* Create an exchange...